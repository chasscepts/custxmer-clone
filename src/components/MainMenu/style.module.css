.menu {
  display: block;
}

.menuItem:not(:first-child) {
  margin-left: 40px;
}

.link {
  display: block;
  border-bottom: 2px solid transparent;
  color: #fff;
  font-size: 16px;
  transition: all 0.3s ease;
}

.link:hover {
  color: #ccc;
}

.link.active {
  border-bottom-color: #fff;
}

.link.active:hover {
  border-bottom-color: #ccc;
}
