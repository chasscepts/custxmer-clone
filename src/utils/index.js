export const paths = {
  HOME: '/',
  ABOUT: '/about',
  SERVICES: '/services',
  CONTACT: '/contact',
};

/**
 * Format path to application specification
 * @param {string} path windows path to fromat
 */
export const formatPath = (path) => {
  if (path === paths.HOME) {
    return '/Home';
  }

  const parts = path.split('/').filter((p) => p);

  return `/${parts.map((p) => `${p[0].toUpperCase()}${p.substring(1).toLowerCase()}`).join('/')}`;
};
