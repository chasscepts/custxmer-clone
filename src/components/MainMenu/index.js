import { Link, useLocation } from 'react-router-dom';
import css from './style.module.css';
import { paths } from '../../utils';

const MainMenu = () => {
  const { pathname } = useLocation();

  return (
    <nav className={css.menu}>
      <ul className="plain-list horizontal">
        {Object.keys(paths).map((key) => (
          <li key={key} className={css.menuItem}>
            <Link
              aria-current={pathname === paths[key]}
              className={`${css.link} ${pathname === paths[key] ? css.active : ''}`}
              to={paths[key]}
            >
              {key}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default MainMenu;
