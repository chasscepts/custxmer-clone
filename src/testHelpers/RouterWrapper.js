import { MemoryRouter } from "react-router";
import PropTypes from 'prop-types';
import { paths } from "../utils";

const RouterWrapper = ({ children }) => (
  <MemoryRouter initialEntries={Object.values(paths)} initialIndex={0}>
    {children}
  </MemoryRouter>
);

RouterWrapper.propTypes = {
  children: PropTypes.node,
};

RouterWrapper.defaultProps = {
  children: null,
};

export default RouterWrapper;
