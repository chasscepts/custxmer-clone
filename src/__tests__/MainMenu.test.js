import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import RouterWrapper from '../testHelpers/RouterWrapper';
import MainMenu from '../components/MainMenu';
import { paths } from '../utils';

describe('Main Menu', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <RouterWrapper>
        <MainMenu />
      </RouterWrapper>
    );
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });

  it('displays all the links', () => {
    render(
      <RouterWrapper>
        <MainMenu />
      </RouterWrapper>
    );

    Object.keys(paths).forEach((key) => expect(screen.getByText(key)).toBeInTheDocument());
  });
});
