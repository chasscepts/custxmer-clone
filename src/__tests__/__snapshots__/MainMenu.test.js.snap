// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Menu matches snapshot 1`] = `
<nav
  className="menu"
>
  <ul
    className="plain-list horizontal"
  >
    <li
      className="menuItem"
    >
      <a
        aria-current={true}
        className="link active"
        href="/"
        onClick={[Function]}
      >
        HOME
      </a>
    </li>
    <li
      className="menuItem"
    >
      <a
        aria-current={false}
        className="link "
        href="/about"
        onClick={[Function]}
      >
        ABOUT
      </a>
    </li>
    <li
      className="menuItem"
    >
      <a
        aria-current={false}
        className="link "
        href="/services"
        onClick={[Function]}
      >
        SERVICES
      </a>
    </li>
    <li
      className="menuItem"
    >
      <a
        aria-current={false}
        className="link "
        href="/contact"
        onClick={[Function]}
      >
        CONTACT
      </a>
    </li>
  </ul>
</nav>
`;
